internal/apps/dop/providers/search/search.go

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2021 Terminus, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package search
+
+import (
+	"context"
+
+	"github.com/erda-project/erda-proto-go/dop/search/pb"
+	"github.com/erda-project/erda/internal/apps/dop/providers/search/handlers"
+	"github.com/erda-project/erda/internal/apps/dop/providers/search/handlers/issue"
+	"github.com/erda-project/erda/internal/apps/dop/services/apierrors"
+	"github.com/erda-project/erda/pkg/common/apis"
+)
+
+type Interface interface {
```

Comment:
this Interface cannot be used in client side.
At client side, the interface implementation is always `p`, and it will use memory invoke, not grpc invoke.

Reply:
adjusted, define a service to impl `searchService`, and provider register the service